compiler/rustc_target/src/spec/targets/wasm32_wasip1_threads.rs

```diff
@@ -16,7 +16,7 @@ pub fn target() -> Target {
 
     options.add_pre_link_args(
         LinkerFlavor::WasmLld(Cc::No),
-        &["--import-memory", "--export-memory", "--shared-memory"],
+        &["--import-memory", "--export-memory", "--shared-memory", "-Wl,--max-memory=1073741824", "-lwasi-emulated-mman"],
```

Comment:
For the Cc::No case the `-Wl,` prefix has to be omitted.

```suggestion
        &["--import-memory", "--export-memory", "--shared-memory", "--max-memory=1073741824", "-lwasi-emulated-mman"],
```